A string of all characters is given including numbers, 
    case 1 : 
        if the string contains even No. of special characters then:
            numbers in the given has to be printed even followed by odd
            
            example : 
                i/p : t9@42&516
                o/p : 492561
                    explanation : this string consists of 2 special chars so 1st occured even followed by 1st occured odd
    case 2 :
        if the string contains odd No. of special characters then:
            numbers in the given has to be printed odd followed by even
            
            example : 
                i/p : 5u6@25g7#@
                o/p : 56527
                    explanation : this string consists of 3 special chars so 1st occured odd followed by 1st occured even

In [3]:
string = input()
even = []
odd = []
c = 0
k = 0
counter = 0
for i in range(len(string)):
    x = ord(string[i])
    if (x > 47 and x < 58) or (x > 64 and x < 91) or (x > 96 and x < 123):
        if x > 47 and x < 58:
            if x % 2 == 0:
                even.append(chr(x))
            else:
                odd.append(chr(x))
        counter += 1
    else:
        c += 1
e = len(even)
o = len(odd)
ec = 0
oc = 0
res = ""
if(c % 2 == 0):
    k = 1
while(counter >= 0):
    if k == 1:
        if ec < e:
            res = res + even[ec]
        ec+=1
        k = 0
    else:
        if oc < o:
            res = res + odd[oc]
        oc+=1
        k = 1
    counter-=1
print(res)

5u6@25g7#@
56527


A string of comma separated numbers. The numbers 5 and 8 are present in it always and assume that 8 always comes after 5.
    case 1:
        we have to concatenate the numbers between 5 and 8 (5 & 8 inclusive)
    case 2:
        The numbers out of 5 and 8 has to be summed
    result produced is the sum of case 1 and case 2
    
    example :
    i/p : 3,2,6,5,1,4,8,9
       ==> 3 + 2 + 6 + 9  = 20
       ==> 5148
       so o/p --> 5418 + 20 = 5168

In [23]:
a = input()
l = list(a.split(","))
k = [int(i) for i in l]
count = 0
s = 0
subcount = ""
while(count <= len(k)):
    if count < len(k):
        if k[count] == 5:
            while(k[count] != 8):
                d = str(k[count])
                subcount = subcount + d
                count+=1
            subcount+='8'
        else:
            s = s + k[count]
    count += 1
print(subcount)
print(s)
s = s + int(subcount)
print("total sum is :",s)

3,4,5,2,7,9,8,3,2
52798
12
total sum is : 52810


A string of different characters is given in which we have to find the largest possible even number after removing the duplicate
elements :
    if we cannot form a even no. we have to return -1
test case :
    1. infosys@3337
    o/p : -1
    2. Hello#81@21349
    o/p : 984312

In [14]:
s = input()
st = set()
for i in range(len(s)):
    x = ord(s[i])
    if x > 47 and x < 58:
        st.add(s[i])
l = []
for i in st:
    l.append(i)
l.sort(reverse = True)
c = 0
d = ""
for i in range(len(l)):
    x = int(i)
    if x % 2 == 0:
        c += 1
    d += l[i]
if(c == 0):
    print(-1)
else:
    k = int(d)
    if(k % 2 == 0):   #if the formed number is even then print the number
        print(k)
    else:
        for i in range(len(l)-1,0,-1):
            if(int(l[i])%2==0):
                x = l[i]
                l.remove(l[i])
                l.append(x)
                break
        for i in l:
            print(int(i),end="")

Hello#81@21349
984312

In [1]:
a = input()
l = []
for i in range(len(a)):
    l.append(a[i])
for i in range(len(l)):
    if i < len(l):
        if l[i] not in ('a','e','i','o','u','A','E','I','O','U'):
            l.pop()
print(l)

vijay
['v', 'i', 'j']


In [8]:
num = input()
rev = num[::-1]
sm = int(num) + int(rev)
rev = str(sm)
rev = rev[::-1]
if int(rev) == sm:
    print(sm)

195


For a given +ve number, Identify the palindrome formed by performing following Operations : 
    * Add number and it's reverse
    * check whether the sum is palindrome or not. If not add the sum and its reverse and repeat the process until a palindrome
      is obtained.
        
    Test Cases : 
        124 ----> o/p : 545
        4 ------> o/p : 8
        195 ----> o/p : 9339

In [1]:
def reverse(num):
    num = num[::-1]
    return int(num)

num = input()
rev = num[::-1]
while(1):
    num = int(num)
    rev = reverse(str(num))
    sm = num + rev
    rev = reverse(str(sm))
    num = sm
    if sm == rev:
        print(sm)
        break

195
9339


You are provided two or more strings, where each string is associated with the number (seperated by :).
If sum of square of digits is even then rotate the string right by one position, and if sum of square of digits 
is odd then rotate the string left by two position.

Accept multiple values in the form of String:Integer seperated by ‘,’
    
    Test Cases : 
        abcde:234,pqrs:246 ------->cdeab spqr
                
Explanation : 
For first teststring, ‘abcde’ associated integer is 234, squaring 4+9+16=29, which is odd, so we rotate string left by two 
positions. For second teststring, ‘spqr’ associated integer is 246, squaring 4+16+36=56, which is even,
so we rotate string right by one position.

In [22]:
def rotate(rot,num):
    if(rot == 1):
        left = num[len(num)-rot:]
        right = num[0:len(num)-rot]
        print(left + right)
    else:
        left = num[0:rot]
        right = num[rot:]
        print(right + left)
    

pairs = input()
l = list(pairs.split(","))
for i in l:
    k = list(i.split(":"))
    num = k[1]
    sm = 0
    for j in num:
        sm = sm + int(j)**2
    if(sm % 2 == 0):
        rotate(1,k[0])
    else:
        rotate(2,k[0])

abcde:234,pqrs:246
cdeab
spqr


Given a special set of numbers and a special sum. Find all those combinations from the given set, equaling the given sum

Test Cases : 

    i/p:-------> list : -1, 1, 0, 0, 2, -2
                 Special Number : 0
    o/p:-------> 3
        
Explanation : Following combinations are satisfying (-1,1,2,-2), (0, 0, 1, -1), (0, 0, -2, 2)

Note : Make combinations with 4 integers only.

In [2]:
import itertools
l=list(input().split(","))
s=int(input())
res=list(itertools.combinations(l,4))
count = 0
for i in res:
    sm = 0
    for j in range(len(i)):
        sm = sm + int(i[j])
    if(sm == s):
        count+=1
print(count)

-1, 1, 0, 0, 2, -2
0
3


You will be given a string of characters and special characters, you task is to reverse the string leaving special characters
in same place.


Input : 
    First Input : String

Output : 
    Special Reverse of string
    
Test Cases :

    i/p : dsd$^f#
    o/p : fds$^d#

In [84]:
s = input()
k = ""
c = 0
l = []
for i in s:
    x = ord(i)
    if (x > 64 and x < 91) or (x > 96 and x < 123):
        k = k + chr(x)
    else:
        l.append(c)
    c += 1
k = k[::-1]
res= ""
c = 0
for i in range(len(s)):
    if i in l:
        res = res + s[i]
    if i not in l:
        if c < len(k):
            res = res + k[c]
            c += 1
print(res)

a$C^d*f%$#
f$d^C*a%$#


For a given list of numbers, find its factors and add the factors. Then if the sum of factors is present in the original list, sort it and print it else print -1.
 
Input : 
    First Input : List of numbers
    Output : List of factors
    
Test Cases :

    i/p ----->1,2,4,7
    o/p ----->[1,4]
    
    i/p ----->2,4
    o/p ----->-1

In [18]:
def factors(num):
    s = 0
    for i in range(1,num+1):
        if num %i == 0:
            s = s + i
    return s

s = input()
l = list(s.split(","))
x = 0
k = []
for i in l:
    x = factors(int(i))
    if str(x) in l:
        k.append(i)
l.clear()
for i in k:
    l.append(int(i))
l.sort()
if len(k) == 0:
    print("-1")
else:
    print(l)

2,4
-1


In [21]:
import itertools
help(itertools)

Help on built-in module itertools:

NAME
    itertools - Functional tools for creating and using iterators.

DESCRIPTION
    Infinite iterators:
    count(start=0, step=1) --> start, start+step, start+2*step, ...
    cycle(p) --> p0, p1, ... plast, p0, p1, ...
    repeat(elem [,n]) --> elem, elem, elem, ... endlessly or up to n times
    
    Iterators terminating on the shortest input sequence:
    accumulate(p[, func]) --> p0, p0+p1, p0+p1+p2
    chain(p, q, ...) --> p0, p1, ... plast, q0, q1, ... 
    chain.from_iterable([p, q, ...]) --> p0, p1, ... plast, q0, q1, ... 
    compress(data, selectors) --> (d[0] if s[0]), (d[1] if s[1]), ...
    dropwhile(pred, seq) --> seq[n], seq[n+1], starting when pred fails
    groupby(iterable[, keyfunc]) --> sub-iterators grouped by value of keyfunc(v)
    filterfalse(pred, seq) --> elements of seq where pred(elem) is False
    islice(seq, [start,] stop [, step]) --> elements from
           seq[start:stop:step]
    starmap(fun, seq) --> fun(*seq

Given a string of random numbers, your job is to find the product of the numbers(one is lesser and one is greater) who is already present in the string. 

For instance, a pronic number is a number which is the product of two consecutive integers, that is, a number of the form 
n(n + 1). Like 6 is the pronic number as 2*3 = 6.

In [20]:
import math
s = input()
k = set()
for i in range(len(s)):
    for j in range(i,len(s)):
        k.add(int(s[i:j+1]))
k = sorted(list(k))
lis = []
for i in k:
    for j in range(1,int(math.sqrt(i))+1):
        if((j)*(j+1) == i):
            lis.append(i)
            break
lis.sort()
if(len(lis) == 0):
    print("-1")
else:
    print(lis)

9135602
[2, 6, 56]


You will be given a number in the form of string, extract out digits at odd places, square & merge them. First 4 digits will be the required OTP.


Input : 

First Input : String

Output : 4 digit OTP

In [9]:
s = input()
sm = ""
for i in range(len(s)):
    if i%2 != 0:
        x = int(s[i])
        sm = sm + str(x * x)
print(sm[:4])

3456
1636


We are given a number n. Our task is to check whether the number is circular prime or not.
Circular Prime : A prime number is said to be a circular prime if after any cyclic permutations of the digits, it remains a prime.
    Test Cases :
    
        i/p : 197
        o/p : Circular Prime
        
        i/p : 54
        o/p : Nope
        
    Explanation ----> 197 971 719 (left rotations) are all primes
                ----> 54 45 (left rotaions) are not primes

In [14]:
def rotate(num,rot=1):
    left = num[0:rot]
    right = num[rot:]
    x = right + left
    return x

def prime(num):
    x = int(num)
    c = 0
    for i in range(2,x//2 + 1):
        if x % i == 0:
            c += 1
    if c == 0:
        return 1
    else:
        return 0

num = input()
l = []
for i in range(len(num)):
    num = rotate(num)
    l.append(num)
sm = 0
for i in l:
    sm = sm + prime(i)
if(sm == len(l)):
    print("circular prime")
else:
    print("Nope")

11
circular prime


Given input of array of string in format employee name:employee number separated by commas.
Find maximum digit in the corresponding employee number which is less or equal to the length of employee name and add that place char into the final string.
Note : If there is no any digit which satisfy the condition then add 'X' into the final string.

Input Format :
First line will be given array of employee name associated with employee number and every employee details will be separated with commas (',').

Output Format : Print final generated password according to given rule above.

Test Case : 
        
        Input :  Robert:36787,Tina:68721,Jo:56389
        Output : tiX
        
Explanation :

    So first employee name and employee number is Robert:36787. Now in the employee number 36787 there is 6 which is maximum and less than or equal to the length of Robert. So we will take 6th character of Robert. So here at 6th position in Robert there is t so we will add t into the final output string.
    
    Same we will do with other employee details. Next one is Tina : 68721 Now in the emp number there is 2 which is maximum and less than or equal to the length of Tina. So we will take 2nd character of Tina. So here at 2nd position in Tina there is i so we will add i into the final output string.

    Next employee is Jo:56389. So in the employee number there is no any digit which is maximum and less than or equal to length of Jo. So in this case we will add X into the final String Output.

    Hence final output string i.e generated password will be tiX.


In [15]:
s = input()
l = list(s.split(","))
print("Password : ",end="")
for i in l:
    k = list(i.split(":"))
    s = len(k[0])
    num = k[1]
    flag = False
    for j in range(s):
        if s == int(num[j-1]):
            print(k[0][s-1],end="")
            flag = True
    if not flag:
        for j in range(s):
            if int(num[j-1]) < s:
                print(k[0][int(num[j-1])],end="")
                break
        else:
            print("X",end="")

Robert:36787,Tina:68721,Jo:56389
Password : tiX

Given an integer limit being the upper limit of the range of interest, implement a function that returns the last 15 palindromes numbers lower or equal to limit as a list sorted ascendingly.

Test Cases : 
        
        i/p :  151
        o/p :  [11, 22, 33, 44, 55, 66, 77, 88, 99, 101, 111, 121, 131, 141, 151]
 
        i/p :  600
        o/p :  [454, 464, 474, 484, 494, 505, 515, 525, 535, 545, 555, 565, 575, 585, 595]


In [22]:
def reverse(num):
    return num[::-1]

s = int(input())
c = 0
l = []
while(c < 15):
    rev = reverse(str(s))
    if(s == int(rev)):
        l.append(s)
        c += 1
    s = s - 1
l.sort()
print(l)

999999
[985589, 986689, 987789, 988889, 989989, 990099, 991199, 992299, 993399, 994499, 995599, 996699, 997799, 998899, 999999]


In [2]:
import itertools
s = input()
n = int(input())
l = list(s.split(","))
m = []
k = []
for i in range(len(l)):
    for j in range(i,len(l)):
        m.append()
    

1,2,3,4,5
1
[('1',), ('2',), ('3',), ('4',), ('5',)]


In [10]:
def dif(l1,l2):
    sm = 0
    for i in l1:
        if i not in l2:
            sm += 1
    return sm

s = input()
s = list(s.split(","))
l1 = s[0]
l2 = s[1]
k = []
m = []
for i in l1:
    k.append(i)
for i in l2:
    m.append(i)
one = dif(k,m)
two = dif(m,k)
print(one+two)

"abc,"ghi"
6


In [4]:
l = list()
for i in range(3):
    for j in range(3):
        n = int(input())
        l.append(n)
for i in range(3):
    for j in range(3):
        print(l[i+j],end="")
    print("")

1
2
3
4
5
6
7
8
9
123
234
345


In [7]:
integer_list = list(map(int, input().split()))

1,2,3,4,5,6


ValueError: invalid literal for int() with base 10: '1,2,3,4,5,6'

In [25]:
def reverse(num):
    return num[::-1]

def larger(num):
    num1 = num
    while(1):
        num = str(num1)
        num = reverse(num)
        if(num1 == int(num)):
            break
        num1 += 1
    return num1


def smaller(num):
    num2 = num
    while(1):
        num = str(num2)
        num = reverse(num)
        if(num2 == int(num)):
            break
        num2 = num2-1
    return num2

        
n = int(input())
large = larger(n)
small = smaller(n)
while(1):
    sm  = large + small
    if(sm == int(reverse(str(sm)))):
        print(sm)
        break
    else:
        n = n - 1
        large = larger(n)
        small = smaller(n)

128
252


In [15]:
def smaller(num):
    num2 = num
    while(1):
        num = str(num2)
        num = reverse(num)
        if(num2 == int(num)):
            return num2
        num2 = num2-1
sm = smaller(66)
print(sm)

66
